firefox-ios/Client/Frontend/Reader/ReaderModeHandlers.swift

```diff
@@ -42,6 +42,11 @@ struct ReaderModeHandlers: ReaderModeHandlersProtocol {
             return GCDWebServerResponse(statusCode: status)
         }
 
+        var readerModeStyle: ReaderModeStyle?
+        ensureMainThread {
+            readerModeStyle = ReaderModeStyle.defaultStyle()
+        }
```

Comment:
Sorry to be nit picky but this now introduces a potential race condition between dispatching to the main thread and `registerHandlerForMethod` finishing below. If `registerHandlerForMethod` manages to finish first it will not have a `readerModeStyle` set. I think line 20 would be a better place to put the `ensureMainThread`.

Reply:
so, to have the entire function running on main thread, ok.

Reply:
it crashes again, even tough we call the method on main thread, the request could execute on another thread and the `ReaderModeStyle.defaultStyle()` it is called again from a background thread. I believe that the best way is to put `ennsureMainThread`  in the declaration `static var defaultSize: ReaderModeFontSize` and in case the contentSizeCategory is nil, it defaults to  `.size6`, if we want the async behavior.